addons/osio-addon/src/main/java/io/fabric8/launcher/osio/jenkins/JenkinsPipelineRegistry.java

```diff
@@ -103,9 +103,19 @@ private JenkinsPipeline readPipeline(Path metadataFile) {
             Path jenkinsfilePath = metadataFile.getParent().resolve("Jenkinsfile");
             boolean suggested = Boolean.valueOf(Objects.toString(metadata.getOrDefault("suggested", "false")));
             boolean techPreview = Boolean.valueOf(Objects.toString(metadata.getOrDefault("tech-preview", "false")));
-            // TODO: Use the real description
-            List<JenkinsPipeline.Stage> stages = (metadata.get("stages") instanceof List ? (List<String>) metadata.get("stages") : Collections.<String>emptyList())
-                    .stream().map(s -> ImmutableStage.of(s, s)).collect(Collectors.toList());
+
+            List<JenkinsPipeline.Stage> stages = (metadata.get("stages") instanceof List ? (List<Object>) metadata.get("stages") : Collections.<Object>emptyList())
+                    .stream().map(s -> {
+                        String stageName = "";
+                        String stageDescription = "";
+                        Map<String, Object> values = (Map<String, Object>)yaml.load(String.valueOf(s));
```

Comment:
I don't understand why are you parsing the YAML again, what is in `s`?

Reply:
We need to parse this again because s contains -
s = {name=Integration Test, description=yyyy creates} this type of YAML structure. or else I can do some multiple string splitting to get the values. 
or you can suggest any other better alternative

Reply:
Are you sure this is not a `java.util.Map` instance?